### PR TITLE
Trip planning initiative

### PR DESCRIPTION
**Pull request by d3mola**
#### What does this PR do?
This PR fetches the travel checklists for open requests

#### Description of Task to be completed?
- Create `GET` travel checklists endpoint
- Write the test for the controller update
- A Travel Admin/Super Admin for a location
-- should be able to edit a previously set travel checklist and the corresponding output of each checklist item
-- should be able to delete a checklist item
-- should be able to create a checklist item

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- Rollback, run database migrations and seed database - `yarn db:rollmigrate`
- Install dependencies and start the server - `make start`
-  [Start](https://github.com/andela/travel_tool_front/pull/107)  the frontend.
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Grab your token from the application tab in Developers tab
- Open `Postman`
- Add your token in headers

<details>
<summary>Retrieve checklist item for a request</summary>
      - Send a `GET` request to `http://localhost:5000/api/v1/checklists?requestId=request-id-6` 
</details>

<details>
<summary>Retrieve checklist percentage completion</summary>
      - Make a get request with the endpoint using http://localhost:5000/api/v1/checklists/completed/qaabjyXYi as an example

</details>

<details>
<summary>Delete a checklist item</summary>
- Send a PUT request to http://localhost:5000/api/v1/checklists/1
        {
             "deleteReason": "deleteReason"
        }

</details>

<details>
<summary>Update a checklist item</summary>
- Send a PUT request to /api/v1/checklist/:id. Using the following request body
Note: /:id (can be 1-8)

         {
          	"name": "Trip Budget",
	        "requiresFiles": true,
	        "resources": [
	        {
	              "link": "linktoreceipt.co",
	              "label": "budget"
	        }
               ]
        }
* You should be able to see updated checklist item
* The response should include a message and a success flag

</details>

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?
[#161044635](https://www.pivotaltracker.com/story/show/161044635)

#### Screenshots (if appropriate)
screen shot 2018-10-11 at 3 39 11 pm

#### Questions:
N/A

### CODE REVIEW

File: src/modules/travelChecklist/__tests__/deleteTravelCheckList.test.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    location: 'Lagos'
+  },
+};
+
+const invalidTokenPayload = {
+  UserInfo: {
+    fullName: 'Jame Jones',
+    email: 'captan.isthmus@andela.com',
+    userId: '--MUyHJmKrxA90lPNQ1FOLN2',
+    // roleId: 53019
```
Comment (ssewilliam) [inline src/modules/travelChecklist/__tests__/deleteTravelCheckList.test.js]: This should also be removed as it is commented code

File: src/middlewares/__mocks__/index.js [hunk 2]
```diff
@@ -39,7 +39,9 @@ const middleware = {
     validateGuestHouse: jest.fn(invokeNextMiddleware),
     validateImage: jest.fn(invokeNextMiddleware),
     getUserId: jest.fn(mockGetUserId),
-    centerExists: jest.fn(invokeNextMiddleware)
+    centerExists: jest.fn(invokeNextMiddleware),
+    validateChecklistQuery: jest.fn(invokeNextMiddleware),
+    // validateRole: jest.fn(invokeNextMiddleware),
```
Comment (ssewilliam) [inline src/middlewares/__mocks__/index.js]: This should be removed as its commented code

File: src/modules/travelChecklist/__tests__/deleteTravelCheckList.test.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+          "msg": "Reason for deletion is required",
+          "param": "deleteReason"
+        }
+      ];
+      
+      request(app)
+      .delete('/api/v1/checklists/4')
+      .set('authorization', token)
+      .end((err, res) => {
+          // console.log('**********************', err);
```
Comment (ssewilliam) [inline src/modules/travelChecklist/__tests__/deleteTravelCheckList.test.js]: This should be removed as it is commented code

File: src/modules/travelChecklist/index.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  authenticate,
+  Validator.validateChecklistQuery,
+  TravelChecklistController.getChecklistsResponse
+);
+
+
+TravelChecklistRouter.delete(
+  '/checklists/:checklistId',
+  authenticate,
+  // Validator.validateRole,
```
Comment (ssewilliam) [inline src/modules/travelChecklist/index.js]: Remove this line

File: src/modules/travelChecklist/index.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  ),
+  deleteChecklistItem,
+  Validator.validateRequest,
+  TravelChecklistController.deleteChecklistItem,
+);
+
+TravelChecklistRouter.post(
+  '/checklists',
+  authenticate,
+  // Validator.validateRole,
```
Comment (ssewilliam) [inline src/modules/travelChecklist/index.js]: remove this line

File: src/modules/travelChecklist/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  ),
+  validateChecklistItem,
+  Validator.validateRequest,
+  TravelChecklistController.createChecklistItem,
+);
+
+TravelChecklistRouter.put(
+  '/checklists/:checklistId',
+  authenticate,
+  // Validator.validateRole,
```
Comment (ssewilliam) [inline src/modules/travelChecklist/index.js]: remove this line

**Comment by Nedson202** [comment]
The changes you requested for have been implemented @ssewilliam

**Pull request merged**